NotificationHubs/src/com/windowsazure/messaging/HttpClientManager.java

```diff
@@ -1,33 +1,77 @@
 package com.windowsazure.messaging;
 
 import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
-import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.client.config.RequestConfig;
 
 public class HttpClientManager {
-	private static CloseableHttpAsyncClient httpAsyncClient;
-	
-	public static CloseableHttpAsyncClient getHttpAsyncClient() {
-		if(httpAsyncClient == null) {
-			synchronized(HttpClientManager.class) {
-				if(httpAsyncClient == null) {
-					CloseableHttpAsyncClient client = HttpAsyncClients.createDefault();
-					client.start();
-					httpAsyncClient = client;	    	   
-				}
-			}
-		}
-		  
-		return httpAsyncClient;
-	}
-		
-	public static void setHttpAsyncClient(CloseableHttpAsyncClient httpAsyncClient) {
-		synchronized(HttpClientManager.class) {
-			if(HttpClientManager.httpAsyncClient == null) {
-				HttpClientManager.httpAsyncClient = httpAsyncClient;
-			}
-			else{
-				throw new RuntimeException("HttpAsyncClient was already set before or default one is being used.");
-			}
-		}
-	}
+
+    private static CloseableHttpAsyncClient httpAsyncClient;
+
+    // A timeout value of zero is interpreted as an infinite timeout.
+    // A negative value is interpreted as undefined (system default).
+
+    // The timeout in milliseconds used when requesting a connection from the connection manager.
+    private static int ConnectionRequestTimeout = -1;
+
+    // The timeout in milliseconds until a connection is established.
+    private static int ConnectionTimeout = -1;
+
+    // The socket timeout in milliseconds, which is the timeout for waiting for data or,
+    // put differently, a maximum period inactivity between two consecutive data packets.
+    private static int SocketTimeout = -1;
```

Comment:
Add documentation to support the `-1` and `0` values such as adding a hyperlink like this `https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/config/RequestConfig.html#getConnectionRequestTimeout()`

Reply:
It was already added at the begin of the class